src/NHibernate.Test/NHSpecificTest/GH0831/FixtureByCode.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+using NHibernate.Cfg.MappingSchema;
+using NHibernate.Mapping.ByCode;
+
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH0831
+{
+	public class ByCodeFixture : TestCaseMappingByCode
+	{
+		private readonly IList<Entity> entities = new List<Entity>
+		{
+			new Entity { EntityValue = 0.5m },
+			new Entity { EntityValue = 1.0m },
+			new Entity { EntityValue = 1.5m },
+			new Entity { EntityValue = 2.0m },
+			new Entity { EntityValue = 2.5m },
+			new Entity { EntityValue = 3.0m }
+		};
+
+		protected override HbmMapping GetMappings()
+		{
+			var mapper = new ModelMapper();
+			mapper.Class<Entity>(rc =>
+			{
+				rc.Id(x => x.Id, m => m.Generator(Generators.GuidComb));
+				rc.Property(x => x.EntityValue);
+			});
+
+			return mapper.CompileMappingForAllExplicitlyAddedEntities();
+		}
+
+		protected override void OnSetUp()
+		{
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				foreach (Entity entity in entities)
+				{
+					session.Save(entity);
+				}
+
+				session.Flush();
+				transaction.Commit();
+			}
+		}
+
+		protected override void OnTearDown()
+		{
+			using (ISession session = OpenSession())
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				session.Delete("from System.Object");
+
+				session.Flush();
+				transaction.Commit();
+			}
+		}
+
+		[Test]
+		public void CanHandleAdd()
+		{
+			CanHandle(e => decimal.Add(e.EntityValue, 2) > 3.0m);
+			CanHandle(e => decimal.Add(2, e.EntityValue) > 3.0m);
+		}
+
+		[Test]
+		public void CanHandleCeiling()
+		{
+			CanHandle(e => decimal.Ceiling(e.EntityValue) > 1.0m);
+		}
+
+		[Test]
+		public void CanHandleCompare()
+		{
+			CanHandle(e => decimal.Compare(e.EntityValue, 1.5m) < 1);
+			CanHandle(e => decimal.Compare(1.0m, e.EntityValue) < 1);
+		}
+
+		[Test]
+		public void CanHandleDivide()
+		{
+			CanHandle(e => decimal.Divide(e.EntityValue, 1.25m) < 1);
```

Comment:
Some databases (Firebird) do not support divide/multiply on decimal. So a new property on `TestDialect` is needed, like `SupportsMultiplicationDecimal`, `true` by default, and overridden to `false` in `FirebirdTestDialect`.

Then divide and multiply tests should `Assume.That(TestDialect.SupportsMultiplicationDecimal, Is.True);`